model/tensorflow_hub/dffml_model_tensorflow_hub/text_classifier.py

```diff
@@ -0,0 +1,420 @@
+        """
+        if not os.path.isfile(
+            os.path.join(self.model_dir_path, "saved_model.pb")
+        ):
+            raise ModelNotTrained("Train model before assessing for accuracy.")
+
+        async for repo in repos:
+            feature_data = repo.features(self.features)
+            df = pd.DataFrame(feature_data, index=[0])
+            predict = await self.prediction_data_generator(np.array(df)[0])
+            all_prob = self._model.predict(predict)
+            max_prob_idx = all_prob.argmax(axis=-1)
+            target = self.parent.config.predict.NAME
+            self.logger.debug(
+                "Predicted probability of {} for {}: {}".format(
+                    self.parent.config.predict.NAME,
+                    np.array(df)[0],
+                    all_prob[0],
+                )
+            )
+
+            repo.predicted(
+                target,
+                self.cids[max_prob_idx[0]],
+                all_prob[0][max_prob_idx[0]],
+            )
+            yield repo
+
+
+@entrypoint("text_classifier")
+class TextClassificationModel(Model):
+    """
+    Implemented using Tensorflow hub pretrained models.
+
+    .. code-block:: console
+
+        $ cat > train.csv << EOF
+        sentence,sentiment
+        Life is good,1
+        This book is amazing,1
+        It's a terrible movie,0
+        Global warming is bad,0
+        EOF
+        $ cat > test.csv << EOF
+        sentence,sentiment
+        I am not feeling good,0
+        Our trip was full of adventures,1
+        EOF
+        $ dffml train \\
+            -model text_classifier \\
+            -model-epochs 30 \\
+            -model-predict sentiment:int:1 \\
+            -model-classifications 0 1  \\
+            -model-clstype int \\
+            -sources f=csv \\
+            -source-filename train.csv \\
+            -model-features \\
+              sentence:str:1 \\
+            -model-model_path "https://tfhub.dev/google/tf2-preview/gnews-swivel-20dim-with-oov/1" \\
+            -model-embedType swivel \\
+            -model-add_layers \\
+            -model-layers "Dense(units=512, activation='relu')" "Dense(units=2, activation='softmax')" \\
+            -log debug
+        $ dffml accuracy \\
+            -model text_classifier \\
+            -model-predict sentiment:int:1 \\
+            -model-classifications 0 1 \\
+            -model-clstype int \\
+            -sources f=csv \\
+            -source-filename test.csv \\
+            -model-features \\
+              sentence:str:1 \\
+            -log critical
+            1.0
+        $ dffml predict all \\
+            -model text_classifier \\
+            -model-predict sentiment:int:1 \\
+            -model-classifications 0 1 \\
+            -model-clstype int \\
+            -sources f=csv \\
+            -source-filename test.csv \\
+            -model-features \\
+              sentence:str:1 \\
+            -log debug
+        [
+        {
```

Comment:
Here's where the indentation issue is coming from, lines 385-415 need to go four spaces to the right.